assets/src/amp-validation/copy-to-clipboard-buttons.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Internal dependencies
+ */
+import { getURLValidationTableRows } from './get-url-validation-table-rows';
+
+/**
+ * Copies any string to the clipboard.
+ *
+ * Note: This only works within a callback responding to a user action.
+ *
+ * @param {string} value Any string value.
+ */
+function copyToClipboard( value ) {
+	const textareaElement = document.createElement( 'textarea' );
+	textareaElement.value = value;
+	textareaElement.style.position = 'absolute';
+	textareaElement.style.left = '-1000%';
+	document.body.appendChild( textareaElement );
+	textareaElement.select();
+	document.execCommand( 'copy' );
+	document.body.removeChild( textareaElement );
+}
+
+/**
+ * Callback when a user clicks a button to copy error details to a clipboard.
+ *
+ * @param {Event} event Click event.
+ */
+function handleCopyToClipboardClick( event ) {
+	// Handle a single error detail button.
+	if ( event.target.classList.contains( 'single-url-detail-copy' ) ) {
+		copyToClipboard( event.target.getAttribute( 'data-error-json' ) );
+		return;
+	}
+
+	// Handle a click on the bulk action button.
+	if ( ! event.target.classList.contains( 'copy-all' ) ) {
+		return;
+	}
+
+	const value = getURLValidationTableRows( { checkedOnly: true } ).map( ( row ) => {
+		const copyButton = row.querySelector( '.single-url-detail-copy' );
```

Comment:
I noticed that if the user changes any of the row's values (without updating the post), the JSON is not updated. For example, updating a validation error to be kept and then copying the JSON for the row does not reflect the updated field value.

Reply:
Fixed in c4e1a6423